crates/core/src/factories/core_fetch/test/response_queue.rs

```diff
@@ -0,0 +1,228 @@
+use super::utils::{random_agent_id, random_id, random_op_id};
+use crate::{
+    default_builder,
+    factories::{
+        core_fetch::{CoreFetch, CoreFetchConfig},
+        test_utils::AgentBuilder,
+    },
+};
+use kitsune2_api::{
+    fetch::{deserialize_ops, Fetch},
+    transport::Transport,
+    MetaOp, OpStore, SpaceId, Timestamp, Url,
+};
+use kitsune2_memory::{Kitsune2MemoryOp, Kitsune2MemoryOpStore};
+use std::{
+    sync::{Arc, Mutex},
+    time::Duration,
+};
+
+#[derive(Debug)]
+pub struct MockTransport {
+    responses_sent: Arc<Mutex<Vec<(Vec<MetaOp>, Url)>>>,
+}
+
+impl MockTransport {
+    fn new() -> Arc<MockTransport> {
+        Arc::new(Self {
+            responses_sent: Arc::new(Mutex::new(Vec::new())),
+        })
+    }
+}
+
+impl Transport for MockTransport {
```

Comment:
Would mockall be appropriate here? Given we just want to work with one method on the type

Reply:
Yes, for sure, I introduced mockall in the other PR and wanted to adapt the existing tests in another PR. But I can do it here already too.

Reply:
No problem if mockall is coming elsewhere. It can be added here later if it's a pain to mix that between PRs

Reply:
I added it now after all.